src/main/java/command/Parser.java

```diff
@@ -0,0 +1,173 @@
+package command;
+
+import task.*;
+
+import java.util.Locale;
+
+import static java.lang.Integer.parseInt;
+
+public class Parser {
+    // Interprets user input
+    Ui ui;
+    TaskList list;
+    public Parser(Ui ui, TaskList list) {
+        this.ui = ui;
+        this.list = list;
+    }
+
+    public void processInput(String input) {
+        String[] inputAnalyzed = input.split(" ");
+        if (input.contains("#")) {
+            ui.showInvalidInputError("Illegal character '#'.");
+            return;
+        }
+        try {
+            switch (inputAnalyzed[0].toLowerCase(Locale.ROOT)) {
+                case "bye":
+                    byeOperation(inputAnalyzed);
+                    break;
+                case "list":
+                    listOperation(inputAnalyzed);
+                    break;
+                case "mark":
+                    markOperation(inputAnalyzed);
+                    break;
+                case "unmark":
+                    unmarkOperation(inputAnalyzed);
+                    break;
+                case "deadline":
+                    ddlOperation(input);
+                    break;
+                case "todo":
+                    todoOperation(input);
+                    break;
+                case "event":
+                    eventOperation(input);
+                    break;
+                case "delete":
+                    deleteOperation(inputAnalyzed);
+                    break;
+                default:
+                    ui.unknownCommand();
+            }
+        } catch (InvalidInputException e) {
+            ui.showInvalidInputError(e.getMessage());
+        } catch (IndexOutOfBoundsException e) {
+            ui.showArrayOutOfBoundsError();
+        } catch (NumberFormatException e) {
+            ui.showNumberFormatError();
```

Comment:
Nice! Catching specific exceptions instead of one catching one general `Exception` will definitely help in debugging